src/geode/mesh/builder/polygonal_surface_builder.cpp

```diff
@@ -167,18 +141,111 @@ namespace geode
+                associate_polygon_vertex_to_vertex(
+                    polygons_around[1], polygon_vertex_id );
+            }
+        }
+
+        polygonal_surface_.update_edge_vertex(
+            { polygon_vertex_id, next_id }, 0, vertex_id );
+        polygonal_surface_.update_edge_vertex(
+            { previous_id, polygon_vertex_id }, 1, vertex_id );
+        polygonal_surface_.remove_isolated_edges();
+        update_polygon_vertex( polygon_vertex, vertex_id );
+    }
+
+    template < index_t dimension >
+    void PolygonalSurfaceBuilder< dimension >::update_polygon_vertices(
+        const std::vector< index_t >& old2new )
+    {
+        for( auto v : geode::Range{ polygonal_surface_.nb_vertices() } )
+        {
+            if( old2new[v] != NO_ID )
+            {
+                auto new_polygon_around =
+                    polygonal_surface_.polygon_around_vertex( old2new[v] );
+                associate_polygon_vertex_to_vertex( new_polygon_around, v );
+            }
+            else
+            {
+                associate_polygon_vertex_to_vertex( PolygonVertex{}, v );
+            }
+        }
+
+        std::vector< bool > polygons_to_delete(
+            polygonal_surface_.nb_polygons(), false );
+
+        for( auto p : geode::Range{ polygonal_surface_.nb_polygons() } )
+        {
+            for( auto v :
+                geode::Range{ polygonal_surface_.nb_polygon_vertices( p ) } )
+            {
+                geode::PolygonVertex id{ p, v };
+                auto new_vertex =
+                    old2new[polygonal_surface_.polygon_vertex( id )];
+                if( new_vertex == geode::NO_ID )
+                {
+                    polygons_to_delete[p] = true;
+                }
+            }
+        }
+        delete_polygons( polygons_to_delete );
+
+        for( auto p : geode::Range{ polygonal_surface_.nb_polygons() } )
+        {
+            for( auto v :
+                geode::Range{ polygonal_surface_.nb_polygon_vertices( p ) } )
+            {
+                geode::PolygonVertex id{ p, v };
+                auto new_vertex =
+                    old2new[polygonal_surface_.polygon_vertex( id )];
+                OPENGEODE_ASSERT( new_vertex != NO_ID,
+                    "[PolygonalSurfaceBuilder::update_polygon_vertices] No "
+                    "more polygons with vertices to delete should remain at "
+                    "this point" );
+                update_polygon_vertex( id, new_vertex );
+            }
+        }
+    }
+
+    template < index_t dimension >
+    void PolygonalSurfaceBuilder< dimension >::update_polygon_vertex(
+        const PolygonVertex& polygon_vertex, index_t vertex_id )
     {
         OPENGEODE_EXCEPTION(
             polygon_vertex.polygon_id < polygonal_surface_.nb_polygons(),
-            "[PolygonalSurfaceBuilder::set_polygon_vertex] Accessing a polygon "
+            "[PolygonalSurfaceBuilder::update_polygon_vertex] Accessing a "
+            "polygon "

```

Comment:
```suggestion
            "polygon that does not exist"
```
and remove the next line